File: day02/star04.py
with open('input.txt') as f:
    lines = f.readlines()

lines = [line.split() for line in lines]

# A = rock
# B = paper
# C = scissors

# X = lose
# Y = draw
# Z = win

play_dict = {'A': {'X': 'S', 'Y': 'R', 'Z': 'P'},  # rock
             'B': {'X': 'R', 'Y': 'P', 'Z': 'S'},  # paper
             'C': {'X': 'P', 'Y': 'S', 'Z': 'R'}}  # scissors

play_score = {'R': 1, 'P': 2, 'S': 3}
game_score = {'X': 0, 'Y': 3, 'Z': 6}

total = 0
for opp, yours in lines:
    score = game_score[yours] + play_score[play_dict[opp][yours]]
    total += score

print(f"{total = }")

File: day02/star04.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <map>

int main()
{
    std::map<std::string, int> game_score {{"A X", 3}, {"A Y", 1}, {"A Z", 2},
                                           {"B X", 1}, {"B Y", 2}, {"B Z", 3},
                                           {"C X", 2}, {"C Y", 3}, {"C Z", 1}};

    std::map<char, int> play_score {{'X', 0}, {'Y', 3}, {'Z', 6}};

    int total = 0;
    std::ifstream f {"input.txt"};
    std::string line;
    while (std::getline(f, line)) {
        total += game_score[line] + play_score[line[2]];
    }

    std::cout << "total = " << total << std::endl;
}
